### PR TITLE
ticdc: adjust TOC and reorganize troubleshooting & faq

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Suggestion from Allen Gao:
- Troubleshoot 部分按照内容分为Troubleshooting 和 FAQ。
- 对于 TiCDC OpenAPI，TiCDC Open Protocol，TiCDC Canal-JSON Protocol 部分建议都放到一个叫“Reference” 的目录下吧。

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by leoppro — APPROVED** [review]
LGTM

File: TOC.md [hunk 1]
```diff
@@ -484,13 +484,15 @@
...
-    - [故障诊断](/ticdc/troubleshoot-ticdc.md)
+    - [故障处理](/ticdc/troubleshoot-ticdc.md)
+    - [常见问题解答](/ticdc/ticdc-faq.md)
     - [监控指标](/ticdc/monitor-ticdc.md)
     - [报警规则](/ticdc/ticdc-alert-rules.md)
-    - [TiCDC Open API](/ticdc/ticdc-open-api.md)
-    - [TiCDC Open Protocol](/ticdc/ticdc-open-protocol.md)
-    - [TiCDC Canal-JSON Protocol](/ticdc/ticdc-canal-json.md)
-    - [TiCDC Avro Protocol](/ticdc/ticdc-avro-protocol.md)
+    - [参考指南]
```
Comment (lilin90) [inline TOC.md:491]:
```suggestion
    - 参考指南
```

**Comment by lilin90** [comment]
I made some changes in https://github.com/pingcap/docs-cn/pull/9945/commits/bef0e0da3ef0d307013d063a13288e55c7b1e1b7, trying to keep the TOC logic consistent between the TiDB TOC and tool's TOC. @shichun-0415 Please help update accordingly for the  English version in https://github.com/pingcap/docs/pull/9166. Thanks!

File: ticdc/ticdc-alert-rules.md [hunk 2]
```diff
@@ -181,7 +181,7 @@ summary: 了解 TiCDC 集群监控报警规则以及处理方法。
 
 * 处理方法：
 
-    MySQL 报错的情况较多，参考[`TiCDC 故障处理`](/ticdc/troubleshoot-ticdc.md)。
+    MySQL 报错的情况较多，参考 [TiCDC 故障处理](/ticdc/troubleshoot-ticdc.md)。
```
Comment (shichun-0415) [inline ticdc/ticdc-alert-rules.md:184]: Similar places in this doc include chapter names by ``, for example, line26, line 40, line 54, line 114. Should I change them as well?
image

**Review by lilin90 — APPROVED** [review]
LGTM

File: TOC.md [hunk 3]
```diff
@@ -484,14 +484,17 @@
...
-    - [将 TiDB 集成到 Confluent Platform](/ticdc/integrate-confluent-using-ticdc.md)
+    - 监控告警
+      - [监控指标](/ticdc/monitor-ticdc.md)
+      - [报警规则](/ticdc/ticdc-alert-rules.md)
+    - [故障处理](/ticdc/troubleshoot-ticdc.md)
+    - 参考指南
+      - [TiCDC Open API](/ticdc/ticdc-open-api.md)
+      - [TiCDC Open Protocol](/ticdc/ticdc-open-protocol.md)
+      - [TiCDC Canal-JSON Protocol](/ticdc/ticdc-canal-json.md)
+      - [TiCDC Avro Protocol](/ticdc/ticdc-avro-protocol.md)
```
Comment (lilin90) [inline TOC.md:495]: Please keep the order of Chinese consistent with that of the English version if no special considerations.
```suggestion
      - [TiCDC Avro Protocol](/ticdc/ticdc-avro-protocol.md)
      - [TiCDC Canal-JSON Protocol](/ticdc/ticdc-canal-json.md)
```

File: ticdc/ticdc-faq.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
+---
+title: 常见问题解答
```
Comment (lilin90) [inline ticdc/ticdc-faq.md:2]:
```suggestion
title: TiCDC 常见问题解答
```

File: ticdc/ticdc-faq.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+summary: 了解 TiCDC 相关的常见问题。
+---
+
+# 常见问题解答
```
Comment (lilin90) [inline ticdc/ticdc-faq.md:6]:
```suggestion
# TiCDC 常见问题解答
```

**Review by lilin90 — COMMENTED** [review]
Please DO NOT use such general wording for titles of a specific tool.

**Comment by lilin90** [comment]
LGTM

**Pull request merged**